scripts/extensions/pool_update.precheck

```diff
@@ -54,18 +70,19 @@ def execute_precheck(session, control_package, yum_conf_file):
     livepatch_dic = {'PATCH_PRECHECK_LIVEPATCH_COMPLETE': 'ok_livepatch_complete', 'PATCH_PRECHECK_LIVEPATCH_INCOMPLETE': 'ok_livepatch_incomplete', 'PATCH_PRECHECK_LIVEPATCH_NOT_APPLICABLE': 'ok'}
     cmd = ['yum', 'install', '-y', '--noplugins', '-c', yum_conf_file, control_package]
     p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True)
-    p.wait()
-    output = p.stdout.read()
+    output, _ = p.communicate()
     xcp.logger.info('pool_update.precheck %r returncode=%r output=%r', cmd, p.returncode, output)
     lines = output.splitlines()
     if p.returncode != 0:
-        errorlines = [l for l in lines if l.startswith('Error: ')]
-        if errorlines:
-            errmsg = errorlines[-1].split(':', 1)[-1].strip()
+        errmsg = re.split(ERROR_MESSAGE_START, output)[-1]
```

Comment:
Update the commits to fix the error message issues (both Xapi ocaml and extension), also add check in precheck extension to ensure there's enough space to apply the update.
For the split issue, will try to improve and update later.